test/local_test.go

```diff
@@ -0,0 +1,79 @@
+package test
+
+import (
+	"errors"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type parsedCRD struct {
+	APIVersion string   `yaml:"apiVersion"`
+	Kind       string   `yaml:"kind"`
+	Metadata   metadata `yaml:"metadata"`
+}
+
+type metadata struct {
+	Annotations annotations `yaml:"annotations"`
+	Name        string      `yaml:"name"`
+}
+
+type annotations struct {
+	EksAmazonawsComRoleArn              string `yaml:"eks.amazonaws.com/role-arn"`
+	EksAmazonawsComStsRegionalEndpoints string `yaml:"eks.amazonaws.com/sts-regional-endpoints"`
+}
+
+func TestServiceAccountsUseStsRegionalEndpoint(t *testing.T) {
+	addonsPath := "../modules/cluster/addons/"
+	addonFiles, err := os.ReadDir(addonsPath)
+	if err != nil {
+		t.Error(err)
+	}
+	for _, v := range addonFiles {
+		fileName := v.Name()
+		if filepath.Ext(fileName) == ".yaml" {
+			crds, err := readManifest(addonsPath + fileName)
+			if err != nil {
+				t.Fatalf("err=%#v\n", err)
+			}
+			for _, v := range crds {
```

Comment:
Maybe overkill, but I'd be tempted to add a `assert.NotEqual(t, 0, len(crds))`

Reply:
I pulled this back out - it's a valid usecase to read a manifest and not have any Kind ServiceAccount